omg/entities/projectile.py

```diff
@@ -82,3 +83,107 @@ def create(cls, init_px, init_py, angle) -> Projectile:
             speed=cls.speed,
             angle=angle,
         )
+
+
+class IceFireFactory(ProjectileFactory):
+    """Convenience class to create IceShard Projectile."""
+
+    image_file = os.path.join(ASSET_DIR, "skills", "IceFire.PNG")
+    scale = 0.05
+    damage = 15
+    speed = 7
+    mana_cost = 20
+
+    @classmethod
+    def create(cls, init_px, init_py, angle) -> Projectile:
+        """Create an IceShard projectile."""
+        return Projectile(
+            name=cls.__name__,
+            image_file=cls.image_file,
+            scale=cls.scale,
+            damage=cls.damage,
+            init_px=init_px,
+            init_py=init_py,
+            speed=cls.speed,
+            angle=angle,
+        )
+
+
+class IceIceFactory(ProjectileFactory):
+    """Convenience class to create IceShard Projectile."""
+
+    image_file = os.path.join(ASSET_DIR, "skills", "IceIce.PNG")
+    scale = 0.05
+    damage = 15
+    speed = 7
+    mana_cost = 20
+
+    @classmethod
+    def create(cls, init_px, init_py, angle) -> Projectile:
+        """Create an IceShard projectile."""
+        return Projectile(
+            name=cls.__name__,
+            image_file=cls.image_file,
+            scale=cls.scale,
+            damage=cls.damage,
+            init_px=init_px,
+            init_py=init_py,
+            speed=cls.speed,
+            angle=angle,
+        )
+
+
+class FireIceFactory(ProjectileFactory):
+    """Convenience class to create IceShard Projectile."""
+
+    image_file = os.path.join(ASSET_DIR, "skills", "FireIce.PNG")
+    scale = 0.05
+    damage = 15
+    speed = 7
+    mana_cost = 20
+
+    @classmethod
+    def create(cls, init_px, init_py, angle) -> Projectile:
+        """Create an IceShard projectile."""
+        return Projectile(
+            name=cls.__name__,
+            image_file=cls.image_file,
+            scale=cls.scale,
+            damage=cls.damage,
+            init_px=init_px,
+            init_py=init_py,
+            speed=cls.speed,
+            angle=angle,
+        )
+
+
+class FireFireFactory(ProjectileFactory):
+    """Convenience class to create IceShard Projectile."""
+
+    image_file = os.path.join(ASSET_DIR, "skills", "FireFire.PNG")
+    scale = 0.05
+    damage = 15
+    speed = 7
+    mana_cost = 20
+
+    @classmethod
+    def create(cls, init_px, init_py, angle) -> Projectile:
+        """Create an IceShard projectile."""
+        return Projectile(
+            name=cls.__name__,
+            image_file=cls.image_file,
+            scale=cls.scale,
+            damage=cls.damage,
+            init_px=init_px,
+            init_py=init_py,
+            speed=cls.speed,
+            angle=angle,
+        )
+
+
+COMBINED_ELEMENT_DICTIONARY = {
```

Comment:
I believe this dictionary can automatically be created somehow by some sort of discovery logic.